src/main/kotlin/no/telenor/kt/restclient/util/util.kt

```diff
@@ -13,10 +14,16 @@ import java.net.URI
 // --- request body utility---
 
 @Suppress("UNNECESSARY_NOT_NULL_ASSERTION")
-fun RestClient.RequestBodySpec.json(value: Any) = this
+fun RequestBodySpec.json(value: Any) = this
 	.body(value)
 	.contentType(MediaType.APPLICATION_JSON)!!
 
+fun RequestBodySpec.header(header: String, value: Any) : RequestBodySpec = this
+	.header(header, value.toString())
+
+fun RequestBodySpec.headers(vararg headersWithValue: Pair<String, Any>) : RequestBodySpec = this
+	.headers { httpHeaders -> headersWithValue.forEach { httpHeaders.add(it.first, it.second.toString()) }}
+
```

Comment:
These header methods should extend `RequestHeaderSpec` and not `RequestBodySpec`. `RequestBodySpec` is used on POST, PATCH, DELETE, etc methods. `RequestHeaderSpec` is returned on GET, OPTIONS and HEAD methods. `RequestBodySpec` extends `RequestHeaderSpec`, so the headers will also become available on `RequestBodySpec` :)

Reply:
oops :)